### PR TITLE
Move data_creator to train and validation

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 1]
```diff
@@ -179,11 +147,14 @@ def find_free_port(self):
         """Finds a free port on the current node."""
         return utils.find_free_port()
 
+    def with_sampler(self, loader):
+        return loader
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:151]: NotImplemented error?

File: pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py [hunk 2]
```diff
@@ -238,13 +218,14 @@ def setup_horovod():
...
     def train(self,
+              data_creator,
               num_steps=None,
               profile=False,
               reduce_results=True,
               info=None):
 
-        success, worker_stats = self._train_epoch(
-            num_steps=num_steps, profile=profile, info=info)
+        success, worker_stats = self._train_epoch(data_creator,
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py:227]: check data_creator is callable

File: pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py [hunk 3]
```diff
@@ -279,7 +261,7 @@ def _train_epoch(self, num_steps=None, profile=False, info=None):
 
         return success, None
 
-    def validate(self, num_steps=None, profile=False, info=None):
+    def validate(self, data_creator, num_steps=None, profile=False, info=None):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py:264]: check data_creator is callable

File: pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py [hunk 4]
```diff
@@ -279,7 +261,7 @@ def _train_epoch(self, num_steps=None, profile=False, info=None):
...
         """Evaluates the model on the validation data set.
 
         Args:
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_horovod_estimator.py:267]: add data_creator description

**Comment by jason-dai** [comment]
LGTM

**Comment by yangw1234** [comment]
jenkins passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3841/
and tested it manually

**Pull request merged**